src/Lynx.purs

```diff
@@ -140,29 +140,29 @@ component =
         <> reflectType x.right
 
   renderTab :: String -> NonEmpty Array (Tab ExprType) -> H.ParentHTML Query (ChildQuery m) ChildSlot m
-  renderTab activeTab tabs = Layout.main_ (Data.Array.fromFoldable $ map (tabSections renderSection renderSequence) tab.sections)
+  renderTab activeTab tabs =
+    Layout.main_ (Data.Array.fromFoldable $ map (tabSections (renderSection Nothing) renderSequence) tab.sections)
     where
     byLink :: Tab ExprType -> Boolean
     byLink { link } = activeTab == link
 
     tab :: Tab ExprType
     tab = fromMaybe (Data.NonEmpty.head tabs) (find byLink tabs)
 
-  renderSection :: Section ExprType -> H.ParentHTML Query (ChildQuery m) ChildSlot m
-  renderSection section =
-    Card.card_
-      $ append [ Format.subHeading_ [ HH.text section.name ] ]
-      $ Data.Array.fromFoldable
-      $ renderField
-      <$> section.fields
+  renderSection :: Maybe Int -> Section ExprType -> H.ParentHTML Query (ChildQuery m) ChildSlot m
+  renderSection index section =
+    Card.card_ $
+      append
+      [ Format.subHeading_ [ HH.text section.name ] ]
+      $ Data.Array.fromFoldable $ renderField (map show index) <$> section.fields
 
   renderSequence :: Sequence ExprType -> H.ParentHTML Query (ChildQuery m) ChildSlot m
   renderSequence sequence =
     Layout.container_
       ( [ Format.subHeading_ [ HH.text sequence.name ] ]
         <> case sequence.values of
-            UserInput sections -> renderSection <$> sections
-            Invalid sections -> renderSection <$> sections
+            UserInput sections -> Data.Array.mapWithIndex (renderSection <<< Just) sections
```

Comment:
Should I be using `FunctorWithIndex` here?

Reply:
I don't think it much matters. They ought to be equivalent if not the same.

Reply:
I ended up changing it anyway